### PR TITLE
Kalamee Q1 2023

### PR DESCRIPTION
**Pull request by DoobesURU**
* Adds scripting files for Age of Kalamee by Semjay and Cyan, Inc.
* Updates xLinkingBookDefs.py with definition for Kalamee.

Note that the second part was a separate commit in case one wants to cherry pick for testing or in case I need to roll back to keep from conflicting with other PRs.

### CODE REVIEW

**Comment by Hoikas** [comment]
`---ERROR Scripts/Python/KalameeResetJC.py:48 ModuleNotFoundError: No module named 'xEnum'---`

**Comment by Hoikas** [comment]
Breaks TL deployment

**Comment by DoobesURU** [comment]
> Breaks TL deployment
> 
> ```
>  Traceback (most recent call last):
>   File "Scripts/Tests/xGamePythonReview.py", line 169, in <module>
>     spec.loader.exec_module(mod)
>   File "<frozen importlib._bootstrap_external>", line 843, in exec_module
>   File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
>   File "/home/runner/work/Plasma/Plasma/Scripts/Python/clftGetPersonalBook.py", line 56, in <module>
>     import xLinkingBookDefs
>   File "/home/runner/work/Plasma/Plasma/Scripts/Python/xLinkingBookDefs.py", line 271, in <module>
>     "ChisoPrenivLower":     ( 0, 1.0, 1.0, NoDRCStamp, BookStart + DRCStampHolder + NoShare + LinkStart + 'xlinkpanelchisolower' + LinkEnd ),
> NameError: name 'BookStart' is not defined
> ```

Oops! `BookStart` vs the proper `BookStart1`. 😊 My bad! New commit should fix that.

**Comment by Hoikas** [comment]
```
(02/11 20:29:44) PythonFile02 - Traceback (most recent call last):
(02/11 20:29:44)   File "xMultiDialogToggle.py", line 134, in OnNotify
(02/11 20:29:44)   File "xMultiDialogToggle.py", line 174, in IStartDialog
(02/11 20:29:44) NameError: name 'true' is not defined. Did you mean: 'True'?
(02/11 20:29:58) xMultiDialogToggle: Dialog SketchbookA goes down
(02/11 20:29:58) PythonFile02 - Traceback (most recent call last):
(02/11 20:29:58)   File "xMultiDialogToggle.py", line 155, in OnGUINotify
(02/11 20:29:58)   File "xMultiDialogToggle.py", line 186, in IQuitDialog
(02/11 20:29:58) NameError: name 'false' is not defined. Did you mean: 'False'?
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+
+
+    def IStartDialog(self, init = 1):
+        PtLoadDialog(Vignette, self.key, self.IGetAgeFilename())
+        if PtIsDialogLoaded(Vignette):
+            PtShowDialog(Vignette)
+            PtDebugPrint(('%s: Dialog %s goes up' % (self.me,
+             Vignette)))
+        if init:
+            PtGetControlEvents(true, self.key)
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:174]:
```suggestion
            PtGetControlEvents(True, self.key)
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+
+
+    def IQuitDialog(self, exit = 1):
+        if ((type(Vignette) != type(None)) and (Vignette != '')):
+            PtHideDialog(Vignette)
+            PtDebugPrint(('%s: Dialog %s goes down' % (self.me,
+             Vignette)))
+        if exit:
+            PtGetControlEvents(false, self.key)
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:186]:
```suggestion
            PtGetControlEvents(False, self.key)
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+        minor = 0
+        self.me = self.__class__.__name__
+        self.VignetteList = []
+        print ('__init__%s v. %d.%d' % (self.me, self.version, minor))
+
+
+
+    def IGetAgeFilename(self):
+        ageInfo = PtGetAgeInfo()
+        if (type(ageInfo) != type(None)):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:99]:
```suggestion
        if ageInfo is not None:
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 4]
```diff
@@ -0,0 +1,187 @@
...
+             Vignette)))
+        if init:
+            PtGetControlEvents(true, self.key)
+            if SingleUser.value:
+                Activate.disable()
+
+
+
+    def IQuitDialog(self, exit = 1):
+        if ((type(Vignette) != type(None)) and (Vignette != '')):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:181]:
```suggestion
        if Vignette:
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+        self.id = 5104999
+        self.version = 1
+        minor = 0
+        self.me = self.__class__.__name__
+        self.VignetteList = []
+        print ('__init__%s v. %d.%d' % (self.me, self.version, minor))
+
+
+
+    def IGetAgeFilename(self):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:97]:
```suggestion
        print ('__init__%s v. %d.%d' % (self.me, self.version, minor))

    def IGetAgeFilename(self):
```
Having so many extra lines of whitespace makes me and my not-huge monitor sad, haha. Please only one line in between blocks.

File: Scripts/Python/xMultiDialogToggle.py [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+        ageInfo = PtGetAgeInfo()
+        if (type(ageInfo) != type(None)):
+            return ageInfo.getAgeFilename()
+        else:
+            return 'GUI'
+
+
+
+    def OnFirstUpdate(self):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:106]:
```suggestion
            return 'GUI'

    def OnFirstUpdate(self):
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+        except:
+            PtDebugPrint(("ERROR: %s.OnFirstUpdate(): Couldn't process dialog list" % self.me))
+            return 
+        for i in range(len(self.VignetteList)):
+            PtLoadDialog(self.VignetteList[i], self.key, self.IGetAgeFilename())
+
+
+
+
+    def __del__(self):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:123]:
```suggestion
            PtLoadDialog(self.VignetteList[i], self.key, self.IGetAgeFilename())

    def __del__(self):
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 8]
```diff
@@ -0,0 +1,187 @@
...
+        for i in range(len(self.VignetteList)):
+            PtUnloadDialog(self.VignetteList[i])
+
+
+
+
+    def OnNotify(self, state, id, events):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:130]:
```suggestion
            PtUnloadDialog(self.VignetteList[i])

    def OnNotify(self, state, id, events):
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 9]
```diff
@@ -0,0 +1,187 @@
...
+        global Vignette
+        if (state and ((id == Activate.id) and PtWasLocallyNotified(self.key))):
+            Vignette = self.VignetteList[0]
+            self.IStartDialog(1)
+
+
+
+    def OnGUINotify(self, id, control, event):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:138]:
```suggestion
            self.IStartDialog(1)

    def OnGUINotify(self, id, control, event):
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 10]
```diff
@@ -0,0 +1,187 @@
...
+                        Vignette = tmpVignette
+                        self.IStartDialog(0)
+                        break
+
+        elif (event == kExitMode):
+            self.IQuitDialog(1)
+
+
+
+    def OnControlKeyEvent(self, controlKey, activeFlag):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:159]:
```suggestion
            self.IQuitDialog(1)

    def OnControlKeyEvent(self, controlKey, activeFlag):
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 11]
```diff
@@ -0,0 +1,187 @@
...
+        PtDebugPrint(('Got controlKey event %d and its activeFlag is %d' % (controlKey,
+         activeFlag)), level=kDebugDumpLevel)
+        if (controlKey == PlasmaControlKeys.kKeyExitMode):
+            self.IQuitDialog(1)
+
+
+
+    def IStartDialog(self, init = 1):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:167]:
```suggestion
            self.IQuitDialog(1)

    def IStartDialog(self, init = 1):
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 12]
```diff
@@ -0,0 +1,187 @@
...
+            PtDebugPrint(('%s: Dialog %s goes up' % (self.me,
+             Vignette)))
+        if init:
+            PtGetControlEvents(true, self.key)
+            if SingleUser.value:
+                Activate.disable()
+
+
+
+    def IQuitDialog(self, exit = 1):
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:180]:
```suggestion
                Activate.disable()

    def IQuitDialog(self, exit = 1):
```

File: Scripts/Python/xMultiDialogToggle.py [hunk 13]
```diff
@@ -0,0 +1,187 @@
...
+        else:
+            return 'GUI'
+
+
+
+    def OnFirstUpdate(self):
+        try:
+            self.VignetteList += Vignettes.value.split(',')
+            for i in range(len(self.VignetteList)):
+                self.VignetteList[i] = self.VignetteList[i].strip()
```
Comment (ZarothYe) [inline Scripts/Python/xMultiDialogToggle.py:110]:
```suggestion
            self.VignetteList = [i.strip() for i in Vignettes.value.split(',')]
```

**Comment by DoobesURU** [comment]
Heh, sorry for the silence. Y'all have been diligent in finding the bugs and I've been slacking at committing them. 😊 

I'll get to this after work today...assuming I'm not too brain-fried by then!

**Comment by Hoikas** [comment]
Thanks for the submission. Unfortunately, at this time, we are not accepting new Ages to this repository. If you are still interested in submitting your new Age, please submit it directly to the Shard(s) that you are interested in it appearing on. Thanks!

**Pull request closed without merging**